src/main/java/UserInterface.java

```diff
@@ -0,0 +1,223 @@
+            return new Deadline(String.valueOf(taskParts.get(0)), String.valueOf(taskParts.get(1)), prepositions);
+        case COMMAND_EVENT:
+            return new Event(String.valueOf(taskParts.get(0)), String.valueOf(taskParts.get(1)), prepositions);
+        }
+        return null;
+    }
+
+    /**
+     * Splits an ArrayList<String> into an ArrayList<StringBuilder>, where each element
+     * can be used to construct a subclass of Task. Use this in conjunction with buildTask to construct a
+     * subclass of Task.
+     * @param input ArrayList of Strings that represent the user's input split by whitespace.
+     * @return an ArrayList<StringBuilder> containing the input split by the '/' character.
+     *         Each element represents a String used to construct the task.
+     *         The last element contains all prepositions used in a single string.
+     *         e.g.: deadline return book /by Sunday /at library returns the following Arraylist:
+     *         {"return book", "Sunday", "Library", "/by/at"}
+     */
+    private ArrayList<StringBuilder> splitTask(ArrayList<String> input) {
+        /* taskPartsIterator increments when a String containing '/' is encountered.
+         * inputIterator increments after each String in input is read.
+         * The loop ends when inputIterator exceeds the size of the ArrayList, indicating that all
+         * Strings in input have been read.
+         */
+        ArrayList<StringBuilder> taskParts = new ArrayList<>();
+        StringBuilder prepositions = new StringBuilder();
+        for (int taskPartsIterator = 0, inputIterator = 0; inputIterator < input.size(); ++taskPartsIterator) {
+            String finalWord = input.get(inputIterator);
+            if (finalWord.startsWith("/")) {
+                prepositions.append(finalWord);
+            }
+            inputIterator++;
+            taskParts.add(new StringBuilder());
+            for (; inputIterator < input.size() && !input.get(inputIterator).startsWith("/"); ++inputIterator) {
+                taskParts.get(taskPartsIterator).append(input.get(inputIterator)).append(" ");
+            }
+        }
+        taskParts.add(prepositions);
+        return taskParts;
+    }
+
+    /**
+     * Prints all tasks stored in memory by addTask(Task) to stdout
+     */
+    private void listTasks() {
+        printDivider();
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 1; i <= this.tasks.size(); i++) {
+            System.out.println(i + "." + this.tasks.get(i - 1));
+        }
+        printDivider();
+    }
+
+    /**
+     * Marks the task selected by the user as done. Tasks are selected by their visual index on the list
+     * (starting from 1, not 0) and not by name.
+     * @param task The index of the task to be marked done.
+     */
+    private void doTask(String task) {
+        if (task == null) {
+            System.out.println(("Error: Task to be done is null."));
```

Comment:
Might want to consider saving line 163, 169, 170, 173 into functions to ensure SLAP.